src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-final class User implements AdvancedUserInterface
+final class User implements UserInterface, EquatableInterface
```

Comment:
Yes, that's what I've been thinking off as well after a discussion with @ro0NL on slack. On one side it feels like this behavior shouldn't be used too often, but I also know that if something _can_ happen, it _will_ happen. 

I just need to find a way to not trigger a deprecation for that.